tests/tests.py

```diff
@@ -1628,6 +1629,31 @@ def test_select_decimal_zero(self):
             result = cur.fetchall()
             self.assertEqual(result[0], expected)
 
+    def test_type_objects(self):
+        with self._connect() as con:
+            cur = con.cursor()
+            cur.execute("""
+select cast(0 as varchar), 
+       cast(1 as binary),
+       cast(2 as int),
+       cast(3 as real),
+       cast(4 as decimal),
+       cast('2005' as datetime),
+       cast('6' as xml)
```

Comment:
Ms sql has a very weird notion of xml. Xml content treated as if it's
wrapped with invisible root tag.
So here it's just a implicit root tag with text '6'.

On Sat, Feb 21, 2015 at 1:25 PM, denisenkom <notifications@github.com>
wrote:

> In tests/tests.py
> <https://github.com/denisenkom/pytds/pull/48#discussion_r25124280>:
>
> > @@ -1628,6 +1629,31 @@ def test_select_decimal_zero(self):
> >              result = cur.fetchall()
> >              self.assertEqual(result[0], expected)
> >
> > +    def test_type_objects(self):
> > +        with self._connect() as con:
> > +            cur = con.cursor()
> > +            cur.execute("""
> > +select cast(0 as varchar),
> > +       cast(1 as binary),
> > +       cast(2 as int),
> > +       cast(3 as real),
> > +       cast(4 as decimal),
> > +       cast('2005' as datetime),
> > +       cast('6' as xml)
>
> this looks like an invalid xml, isn't it failing?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/denisenkom/pytds/pull/48/files#r25124280>.
>



-- 
Regards,
  Valerii Lashmanov